gym/envs/box2d/car_racing.py

```diff
@@ -517,10 +517,11 @@ def reset(
             self.render()
         return self.step(None)[0], {}
 
-    def step(self, action: Union[np.ndarray, int]):
+    def step(self, action: Optional[Union[np.ndarray, int]]):
```

Comment:
While I agree this is the correct typing, I don't think we want to encourage users to do this. Could we use type ignore on the reset

Reply:
This is true. I would update it.